google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/Internal.java

```diff
@@ -30,8 +30,10 @@
   private FirestoreImpl firestore;
 
   @InternalApi
-  public Internal(FirestoreImpl firestore) {
-    this.firestore = firestore;
+  public Internal(Firestore firestore) {
```

Comment:
This constructor is still marked `@InternalApi` so it shouldn't b used by other code. Is this change realy needed?

Reply:
I have a meeting with the team that requested this on Monday, after which I can follow up. The team is Internal, so they are "allowed" to use these APIs.

Reply:
What's weird is that you're doing this to avoid exposing an internal class in the API, but the API is internal so why not just leave it as is?

Reply:
FirestoreImpl is not constructible outside of our package.

Reply:
Dropping this PR for now. Cloud Frameworks team will present their own solution.